extensions/vertx-web/deployment/src/test/java/io/quarkus/vertx/web/reactive/RequestContextPropagationTest.java

```diff
@@ -0,0 +1,78 @@
+package io.quarkus.vertx.web.reactive;
+
+import static io.restassured.RestAssured.get;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.annotation.PreDestroy;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.vertx.web.Route;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+public class RequestContextPropagationTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(MyRoutes.class, Ping.class));
+
+    @Test
+    public void test() {
+        get("/ping?val=bar").then().statusCode(200).body(is("foo_bar"));
+        assertTrue(Ping.DESTROYED.get());
```

Comment:
This is racey.

As far as the client is concerned the request is complete when the last bit of data is received from the wire. This happens in the vert.x end() call, and at that point the bean has not been destroyed, as the destruction happens after the data is sent. Usually the test will pass because destroying the bean happens straight after the data is sent, and sending the data is relatively slow, but occasionally it will fail. This is a very common cause of intermittent CI failures. 

If you change the AtomicBoolean to a CompletableFuture and wait a few seconds for the value this will fix it.

Reply:
I see. Thanks for spotting this!